src/server/cluster/cluster_family.cc

```diff
@@ -630,39 +630,60 @@ void ClusterFamily::DflyClusterStartSlotMigration(CmdArgList args, ConnectionCon
   return cntx->SendOk();
 }
 
+static std::string_view state_to_str(ClusterSlotMigration::State state) {
+  switch (state) {
+    case ClusterSlotMigration::C_NO_STATE:
+      return "NO_STATE"sv;
+    case ClusterSlotMigration::C_CONNECTING:
+      return "CONNECTING"sv;
+    case ClusterSlotMigration::C_FULL_SYNC:
+      return "FULL_SYNC"sv;
+    case ClusterSlotMigration::C_STABLE_SYNC:
+      return "STABLE_SYNC"sv;
+  }
+  assert(false);
+  return "UNDEFINED_STATE"sv;
+}
+
 void ClusterFamily::DflySlotMigrationStatus(CmdArgList args, ConnectionContext* cntx) {
   CmdArgParser parser(args);
-  auto [host_ip, port] = parser.Next<std::string_view, uint16_t>();
+  auto* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
 
-  if (auto err = parser.Error(); err)
-    return cntx->SendError(err->MakeReply());
+  if (parser.HasNext()) {
+    auto [host_ip, port] = parser.Next<std::string_view, uint16_t>();
+    if (auto err = parser.Error(); err)
+      return rb->SendError(err->MakeReply());
 
-  auto state = [&] {
     lock_guard lk(migration_mu_);
+    // find incoming slot migration
     for (const auto& m : migrations_jobs_) {
```

Comment:
I think it might be good to add the incoming / outgoing name to the migrations_jobs_ / migration_infos_ var names